features/step_definitions/logging.rb

```diff
@@ -362,6 +362,9 @@
     end
       step %Q/logging collector name is stored in the :collector_name clipboard/
     unless cluster_logging('instance').collection_spec(cached: true).nil?
+      if daemon_set(collector_name).exists?
+        @result = admin.cli_exec(:delete, object_type: 'daemonset', object_name_or_id: collector_name, n: 'openshift-logging')
+      end
```

Comment:
You can try the step `Given admin ensures "xxx_name_xxx" xxx_resource_type_xxx is delete` to simplify the logic of `check xxx exist, then run the delete command, then wait/check it's gone`